### PR TITLE
Add initial DragonFly BSD support

### PR DESCRIPTION
**Pull request by tuxillo**
This is the first approach for getting this building/working in DragonFly BSD. It is a requirement of lima-vm (https://github.com/lima-vm/lima/issues/892)

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
Thanks, but please sign the commit for DCO
https://github.com/apps/dco

(run `git commit -a -s --amend`, and make sure that the `Signed-off-by: NAME <EMAIL>` line with your real name is included in the commit message)

File: devices/mknod_freebsd.go [hunk 1]
```diff
@@ -1,5 +1,5 @@
-//go:build freebsd
-// +build freebsd
+//go:build freebsd dragonfly
```
Comment (AkihiroSuda) [inline devices/mknod_freebsd.go:1]:
```suggestion
//go:build freebsd || dragonfly
```

**Comment by AkihiroSuda** [comment]
CI is failing for FUSE (you may just disable it for dragonfly)

Also please consider squashing the commits

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Review by samuelkarp — APPROVED** [review]

**Pull request merged**